core/src/main/scala/io/projectglow/sql/expressions/LogisticRegressionExpr.scala

```diff
@@ -19,29 +19,102 @@ package io.projectglow.sql.expressions
 import scala.collection.mutable
 import scala.util.hashing.MurmurHash3
 
-import breeze.linalg.{DenseMatrix => BreezeDenseMatrix, DenseVector}
+import breeze.linalg.{DenseVector, DenseMatrix => BreezeDenseMatrix}
+import org.apache.spark.TaskContext
 import org.apache.spark.ml.linalg.DenseMatrix
 import org.apache.spark.sql.SQLUtils
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
-import org.apache.spark.sql.catalyst.expressions.codegen.CodegenFallback
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, CodegenFallback, ExprCode}
 import org.apache.spark.sql.catalyst.expressions.{Expression, ImplicitCastInputTypes, QuaternaryExpression}
 import org.apache.spark.sql.catalyst.util.ArrayData
 import org.apache.spark.sql.types.{ArrayType, DataType, DoubleType, StringType}
 import org.apache.spark.unsafe.types.UTF8String
 
+class LogisticRegressionState(testStr: String) {
+  val logitTest = LogisticRegressionGwas
+    .logitTests
+    .getOrElse(
+      testStr,
+      throw new IllegalArgumentException(
+        s"Supported tests are currently: ${LogisticRegressionGwas.logitTests.keys.mkString(", ")}")
+    )
+  var fitState: logitTest.FitState = _
```

Comment:
Comment that only one of `fitState` or `nullFitMap` should be non-`null`/empty respectively.